### PR TITLE
improved the string formatting syntax

### PR DESCRIPTION
**Pull request by sayantan1410**
I have changed the format of the strings in the print statement from old percentage version to the new format. 

Contributes to #4136

### CODE REVIEW

File: tests/c_api_test/test_.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ def test_booster():
             ctypes.byref(out_len),
             result.ctypes.data_as(ctypes.POINTER(ctypes.c_double)))
         if i % 10 == 0:
-            print('%d iteration test AUC %f' % (i, result[0]))
+            print(f'{int(i)} interation test AUC {result[0]:6f}')
```
Comment (StrikerRUS) [inline tests/c_api_test/test_.py:237]: `int` is excess here.
Did you want to specify ``width`` or `precision` for AUC?
```
f'{value:{width}.{precision}f}'
```
```suggestion
            print(f'{i} interation test AUC {result[0]:.6f}')
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
@sayantan1410 Thanks for your contribution!
Please update this line as well 
https://github.com/microsoft/LightGBM/blob/adf36d766afb6261276d0624f4f5b216d51880c8/tests/c_api_test/test_.py#L32
in a similar manner
https://github.com/microsoft/LightGBM/blob/adf36d766afb6261276d0624f4f5b216d51880c8/helpers/parameter_generator.py#L222-L223

File: tests/c_api_test/test_.py (hunk 1 above)
Comment (sayantan1410) [reply tests/c_api_test/test_.py:237]: Thank you so much for the review, I missed that line number 32 required some changes, and for the second case I the value was printed as float that is why I typecasted into int but from your suggestion I understand where I am going wrong. 
Will submit a Pull request after updating these soon.

**Comment by jameslamb** [comment]
Thanks for the help @sayantan1410 ! Please note I've added the following to the description of this pull request:

> Contributes to #4136

Mentioning any prior discussions about a change helps us, as reviewers, to more quickly understand what your pull request is trying to accomplish. It also helps us to trace back through the project's history and find related discussions.

File: tests/c_api_test/test_.py (hunk 1 above)
Comment (sayantan1410) [reply tests/c_api_test/test_.py]: Hello, I have made the changes that you have requested. Hope I haven't messed up. @StrikerRUS

**Comment by sayantan1410** [comment]
Okay thank you so much. @jameslamb

**Comment by jameslamb** [comment]
Sorry for the inconvenience, but we recently merged a fix for some CI issues (#4288 ). Whenever you have time, could you please update this branch to the latest `master`? That should fix the failing builds.

**Comment by sayantan1410** [comment]
I will surely do that, there is absolutely no problem @jameslamb

**Comment by sayantan1410** [comment]
Hey, I have updated my branch with the master branch. Let me know if I have done everything correctly or something is needed to be changed or updated. @jameslamb

File: tests/c_api_test/test_.py [hunk 2]
```diff
@@ -29,7 +29,8 @@ def find_lib_path():
     lib_path = [p for p in dll_path if os.path.exists(p) and os.path.isfile(p)]
     if not lib_path:
         dll_path = [os.path.realpath(p) for p in dll_path]
-        raise Exception('Cannot find lightgbm library file in following paths:\n' + '\n'.join(dll_path))
+        dll_path_joined = '\n\n'.join(dll_path)
+        raise Exception(f'Cannot find lightgbm library file in following paths: {dll_path_joined}')
```
Comment (StrikerRUS) [inline tests/c_api_test/test_.py:33]: The first `\n` was to separate the first sentence from the list of paths and items inside the list were combined with only one `\n` not two.
```suggestion
        dll_path_joined = '\n'.join(dll_path)
        raise Exception(f'Cannot find lightgbm library file in following paths:\n{dll_path_joined}')
```

File: tests/c_api_test/test_.py [hunk 3]
```diff
@@ -234,7 +235,7 @@ def test_booster():
             ctypes.byref(out_len),
             result.ctypes.data_as(ctypes.POINTER(ctypes.c_double)))
         if i % 10 == 0:
-            print('%d iteration test AUC %f' % (i, result[0]))
+            print(f'{i} interation test AUC {result[0]:.6f}')
```
Comment (StrikerRUS) [inline tests/c_api_test/test_.py:238]:
```suggestion
            print(f'{i} iteration test AUC {result[0]:.6f}')
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]
@sayantan1410 Thanks for adding line 32 into this PR. Please take a look at my new comments below.

Also, looks like there was a merging conflict - this PR shouldn't contain any files other than `tests/c_api_test/test_.py`. Could you please fix this conflict?

File: tests/c_api_test/test_.py (hunk 2 above)
Comment (sayantan1410) [reply tests/c_api_test/test_.py:33]: Okay i will update the '/n' part and should I take back the commits that committed other files and make a fresh commit ? probably that is the way to do it. @StrikerRUS

File: tests/c_api_test/test_.py (hunk 3 above)
Comment (sayantan1410) [reply tests/c_api_test/test_.py:238]: I am so ashamed of this kind of errors, really sorry, will surely fix it. @StrikerRUS

**Comment by sayantan1410** [comment]
Hey, Because of some issue there was lots of errors in the check that's why I deleted all the commits but that for some reason closed the pull request, I didn't know that this happens, I am really sorry, Can I open a new pull request, I have everything figured out for the issue. Sorry for the inconvenience, I am ashamed I did this but it will be great if I solve the issue from another pull request. Actually I am new to open source contribution that is why I messed it up.  @jameslamb

**Comment by jameslamb** [comment]
Yes, please open a new pull request and we'd be happy to review it. No worries! `@` me if you need any help.

**Comment by sayantan1410** [comment]
Thank you so much !!!

File: tests/c_api_test/test_.py (hunk 3 above)
Comment (StrikerRUS) [reply tests/c_api_test/test_.py]: Oh, please don't be ashamed! We appreciate your help!

**Pull request closed without merging**